File: BinaryTree/Leetcode99.java
public class Leetcode99 {
    public void recoverTree(TreeNode root) {
        List<TreeNode> list = new ArrayList<>();
        inorder(root,list);
        TreeNode first = null,second = null;
        for(int i = 1;i < list.size();i++){
            if(list.get(i-1) < list.get(i))continue;
            if(first == null)first = list.get(i-1); 
            second =list.get(i);
        }
        if(first == null)return;
        int temp = first.val;
        first.val =second.val;
        second.val = temp;
    }
    //first we inorder traverse the BST
    // Then we will get a array that only havs two element not in order
    public void inorder(TreeNode root, List<TreeNode> list){
        if(root == null)return;
        inorder(root.left,list);
        list.add(root);
        inorder(root.right,list);
    }
}

File: BinarySearchTree/Python/Leetcode99.py
class Leetcode99:
    def recoverTree(self, root: Optional[TreeNode]) -> None:
        """
        Do not return anything, modify root in-place instead.
        """
        f = s = -1
        tree = []
        self.inorder(root, tree)
        for i in range(1, len(tree)):
            if tree[i-1].val < tree[i].val: continue
            if f == -1: f = i - 1
            s = i
        temp = tree[s].val
        tree[s].val = tree[f].val
        tree[f].val = temp


    
    def inorder(self, node: TreeNode, tree: list) -> None:
        if node is None:
            return
        self.inorder(node.left, tree)
        tree.append(node)
        self.inorder(node.right, tree)
